docs/developer_guide/building.md

```diff
@@ -0,0 +1,201 @@
+# Building geOrchestra Gateway
+
+This document describes how to build the geOrchestra Gateway from source and set up a development environment.
+
+!!! note "Development Environment"
+    The repository contains `docker-compose.yml` and `docker-compose-preauth.yaml` files in the root directory. These are **for development purposes only** and should not be used for production deployments. They provide a convenient way to run a local development environment with all required services.
+
+## Prerequisites
+
+Before you can build the geOrchestra Gateway, you need:
+
+- Java 21 or higher
+- Maven 3.6.3 or higher
+- Git
+
+## Clone the Repository
+
+First, clone the repository:
+
+```bash
+git clone https://github.com/georchestra/georchestra-gateway.git
+cd georchestra-gateway
+```
+
+## Build Commands
+
+geOrchestra Gateway uses Maven as its build system. Here are the main build commands:
+
+### Basic Build
+
+To build the project:
+
+```bash
+./mvnw clean install
+```
+
+Or using the Makefile:
+
+```bash
+make install
+```
+
+This will:
+1. Clean any previous build artifacts
+2. Compile the code
+3. Run the tests
+4. Install the artifacts to your local Maven repository
+
+### Skip Tests
+
+To build without running tests:
+
+```bash
+./mvnw clean install -DskipTests
+```
+
+Or using the Makefile:
+
+```bash
+make install
+```
+
+### Run Tests
+
+To run the tests:
+
+```bash
+./mvnw verify
+```
+
+Or using the Makefile:
+
+```bash
+make test
+```
+
+### Run a Single Test
+
+To run a single test class:
+
+```bash
+./mvnw test -pl :georchestra-gateway -ntp -Dtest=TestClassName
+```
+
+For integration tests:
+
+```bash
+./mvnw verify -pl :georchestra-gateway -ntp -Dit.test=ITClassName
+```
+
+### Build Docker Image
+
+To build the Docker image:
+
+```bash
+./mvnw package -f gateway/ -Pdocker -ntp -DskipTests
+```
+
+Or using the Makefile:
+
+```bash
+make docker
+```
+
+### Format Code
+
+To format the code according to the project's style guidelines:
+
+```bash
+./mvnw formatter:format -pl :georchestra-gateway
+```
+
+### Validate Formatting
+
+To validate that the code follows the project's style guidelines:
+
+```bash
+./mvnw formatter:validate -pl :georchestra-gateway
+```
+
+## Development Environment
```

Comment:
Maybe some info are missing. Something I usually like is also a simple _Run it in the commandline_ instruction 
(`./mvnw spring-boot:run -Dspring-boot.run.profiles=json-logs,dev -pl :georchestra-gateway`)
which also shows how to target specific profiles. 
We can also provide the `SPRING_PROFILES_ACTIVE=json-logs,dev` way)

Reply:
* [fix](https://github.com/georchestra/georchestra-gateway/compare/790b7da38a57ecbbade37684744ca04cc604fce0..71b81fb7371059f9329481d46d8e0d9bda2157d9#diff-9ead6f7a795a284273759f052cb49ffcdd3aba154260dff5dcc756fb671ed766R133)
* live: https://groldan.github.io/georchestra-gateway/developer_guide/building/#running-from-the-command-line